### PR TITLE
Updated default access token binding type of system apps to cookie

### PR DESCRIPTION
**Pull request by mpmadhavig**
## Purpose
With the new IS 6.0.0 release the default token binding type of system apps (console app and myaccount app) are changed to cookie based. This change will automate the migration process of changing default token binding type from SSO-based binding to cookie-based binding.

### CODE REVIEW

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java [hunk 1]
```diff
@@ -30,18 +30,18 @@
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
-/**
- * Migration implementation for updating the redirect URLs of System Applications.
- */
```
Comment (ashendes) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:35]: Javadoc comment is required.

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java [hunk 2]
```diff
@@ -53,17 +53,31 @@ public void dryRun() throws MigrationClientException {
...
+
+            JDBCPersistenceUtil.commitTransaction(connection);
+            connection.setAutoCommit(autoCommitStatus);
+        } catch (SQLException e) {
+            throw new MigrationClientException("An error occurred while updating system application", e);
+        }
+    }
+
+    private void updateRedirectUrl(Connection connection, boolean autoCommitStatus) throws MigrationClientException {
+        log.info("Started updating System application redirect URLs.");
```
Comment (ashendes) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:80]: Add a new line after method signature.

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java [hunk 3]
```diff
@@ -78,16 +92,44 @@ public void migrate() throws MigrationClientException {
             } catch (SQLException e) {
                 JDBCPersistenceUtil.rollbackTransaction(connection);
                 connection.setAutoCommit(autoCommitStatus);
-                throw new MigrationClientException("An error occurred while updating system application" +
+                throw new MigrationClientException("An error occurred while updating System application" +
                         " redirect URLs.", e);
             }
+        } catch (SQLException e) {
```
Comment (ashendes) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:101]: Is the outer try-catch block necessary?

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java [hunk 4]
```diff
@@ -78,16 +92,44 @@ public void migrate() throws MigrationClientException {
...
+            throw new MigrationClientException("An error occurred while updating System application" +
+                    " redirect URLs.", e);
+        }
+        log.info("System application redirect url migration complete.");
+    }
 
-            JDBCPersistenceUtil.commitTransaction(connection);
-            connection.setAutoCommit(autoCommitStatus);
+    private void updateAccessTokenBindingType(Connection connection, boolean autoCommitStatus) throws MigrationClientException {
```
Comment (ashendes) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:108]: Line length longer than 120 characters. Pls fix.

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java (hunk 3 above)
Comment (mpmadhavig) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:102]: Inside the inner catch block there's a connection.setAutocommit command which can cause an SQL exception.

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java [hunk 5]
```diff
@@ -78,16 +96,45 @@ public void migrate() throws MigrationClientException {
...
+        try {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_ACCESS_TOKEN_BINDING_TYPE)) {
+                if (StringUtils.isNotBlank(accessTokenBindingType)) {
+                    preparedStatement.setString(1, accessTokenBindingType);
+                    preparedStatement.setString(2, "tokenBindingType");
+                    preparedStatement.setString(3, "CONSOLE");
+                    preparedStatement.executeUpdate();
+
+                    preparedStatement.setString(1, accessTokenBindingType);
+                    preparedStatement.setString(2, "tokenBindingType");
```
Comment (ashendes) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v600/migrator/SystemAppMigrator.java:123]: These two lines are redundant.

**Pull request merged**